src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -965,6 +1048,15 @@ void VM_Version::get_processor_features() {
     _features &= ~CPU_AVX512_IFMA;
   }
 
+  // Currently APX support is only enabled for targets supporting AVX512VL feature.
+  if (UseAPX && (!supports_apx_f() || !supports_avx512vl())) {
```

Comment:
This code should be after UseAVX checks.

Reply:
Its purposefully placed after modifications to CPU_* features flags if user explicitly sets UseAVX < 3.

Reply:
Got it. I missed that we have separate UseAVX checks for <3 and < 2 and < 1.